lib/aws_one_click_staging/aws_warrior.rb

```diff
@@ -104,12 +104,23 @@ def setup_aws_credentials config
       missing = CREDENTIAL_KEYS.select do |key|
         !config[key]
       end
+
+      #check if there are some credentials on the machine
+      begin
+        identity = Aws::STS::Client.new().get_caller_identity
+      rescue => e
+        p "no credentials"
+      else
+        p "the credentials are:"
+        p identity
```

Comment:
Please drop the else branch and put a more specific class in the rescue if there is previous failed job history to use.